api/src/main/java/org/ccci/idm/user/DefaultUserManager.java

```diff
@@ -184,8 +185,8 @@ protected void setNewUserDefaults(final User user) throws UserException {
         // generate a guid for the user if there isn't a valid one already set
         int count = 0;
         String guid = user.getGuid();
-        while (!StringUtils.hasText(guid) || this.doesGuidExist(guid) || this.doesRelayGuidExist(guid) || this
-                .doesTheKeyGuidExist(guid)) {
+        while (!UserUtil.isValidGuid(guid) || doesGuidExist(guid) || doesRelayGuidExist(guid) ||
```

Comment:
this was just somewhere that it made sense to utilize it

Reply:
Is it ever possible that https://github.com/CruGlobal/idm-user-management/pull/107/commits/249db346dc8fc711f67647946b26131c9b8e67dd#diff-b9acd95562384f16b5e2b1d8eaec4c78R187 would produce an invalid guid?

Reply:
the `User` object passed in is created by other applications that utilize this library, so maybe, maybe not.